slides/k8s/setup-selfhosted.md

```diff
@@ -0,0 +1,108 @@
+# Kubernetes distributions and installers
+
+- There are [countless](https://kubernetes.io/docs/setup/pick-right-solution/) distributions available
+
+- We can't review them all
+
+- We're just going to explore a few options
+
+---
+
+## kops
+
+- Deploys Kubernetes using cloud infrastructure
+
+  (supports AWS, GCE, Digital Ocean ...)
+
+- Leverages special cloud features when possible
+
+  (e.g. Auto Scaling Groups ...)
+
+---
+
+## kubeadm
+
+- Provisions Kubernetes nodes on top of existing machines
+
+- `kubeadm init` to provision a single-node control plane
+
+- `kubeadm join` to join a node to the cluster
+
+- Supports HA control plane [with some extra steps](https://kubernetes.io/docs/setup/independent/high-availability/) 
+
+---
+
+## Kubespray
+
+- Based on Ansible
+
+- Works on bare metal and cloud infrastructure
+
+  (good for hybrid deployments)
+
+- The expert says: ultra flexible; slow; complex
+
+---
+
+## RKE (Rancher Kubernetes Engine)
+
+- Opinionated installer with low requirements
+
+- Requires a set of machines with Docker + SSH access
+
+- Supports highly available etcd and control plane
+
+- The expert says: fast; maintenance can be tricky
+
+---
+
+## Terraform + kubeadm
+
+- Sometimes it is necessary to build a custom solution
+
+- Example use case: 
+
+  - deploying Kubernetes on OpenStack
+
+  - ... with highly available control plane
+
+  - ... and Cloud Controller Manager integration
+
+- Solution: Terraform + kubeadm (kubeadm driven by remote-exec)
+
+  - [GitHub repository](https://github.com/enix/terraform-openstack-kubernetes)
+
+  - [Blog post (in French)](https://enix.io/fr/blog/deployer-kubernetes-1-13-sur-openstack-grace-a-terraform/)
+
+---
+
+## And many more ...
+
+- Docker Enterprise Edition
+
+- Pivotal Container Service (PKS)
+
+- Tectonic by CoreOS
```

Comment:
Tectonic is being combined with OpenShift though: https://coreos.com/blog/coreos-tech-to-combine-with-red-hat-openshift

Reply:
Do you think it needs to be worded differently? My intent is just to namedrop a few other solutions, but I won't even be able to say how they're different (except in a very superficial way, and even then!)

Reply:
I guess what I mean is, I think it's not really going to be a choice pretty soon?

Reply:
Definitely not a blocker for this entire PR =)

Reply:
I'll be tracking this in #454 !